app/components/article-content.js

```diff
@@ -607,67 +608,66 @@ export default Component.extend(
 
     /**
      * Injects an affiliate unit into the article content
-     * @param {string} title
      */
-    handleBigAffiliateUnit(title) {
-      const unit = this.affiliateSlots.getBigUnitOnPage(title);
-
-      if (typeof unit === 'undefined') {
-        // There's no unit to display (not an error)
-        return;
-      }
-
-      // search for second section
-      const h2Elements = this.element.querySelectorAll('h2[section]');
+    handleBigAffiliateUnit() {
+      this.affiliateSlots
+        .fetchUnitForPage(this.id, true)
+        .then(unit => {
+          if (typeof unit === 'undefined') {
+            // There's no unit to display (not an error)
+            return;
+          }
 
-      if (h2Elements[1]) {
-        const unitPlaceholder = document.createElement('div');
-        const unitWrapper = document.createElement('div');
+          // search for second section
+          const h2Elements = this.element.querySelectorAll('h2[section]');
```

Comment:
Any chance this can be null?

Reply:
> A non-live NodeList containing one Element object for each element that matches at least one of the specified selectors or an empty NodeList in case of no matches.

Looks like we are okay 👍